File: api/src/main/java/project/price_it/dto/user/UserDto.java
package project.price_it.dto.user;

import lombok.*;
import project.price_it.entity.UserEntity;

@NoArgsConstructor
@AllArgsConstructor
@ToString
@Setter
@Getter
@Builder
public class UserDto {
    private Long id;

    private String email;
    private String password;

    private String name = "example";

    private Double lat = 0.0;
    private Double lng = 0.0;

    private int point = 0;

    public UserEntity toEntity() {
        return UserEntity.builder()
                .id(id)
                .name(name)
                .lat(lat)
                .lng(lng)
                .point(point)
                .email(email)
                .password(password)
                .build();
    }

    public static UserDto fromEntity(UserEntity user) {
        if (user == null) return null;

        return UserDto.builder()
                .id(user.getId())
                .name(user.getName())
                .point(user.getPoint())
                .email(user.getEmail())
                .build();
    }
}

File: api/src/main/java/project/price_it/dto/user/UserDto.kt
package project.price_it.dto.user

import project.price_it.entity.UserEntity

data class UserDto(
        val id: Long? = null,
        val email: String,
        val password: String,
        val name: String = "example",
        val lat: Double = 0.0,
        val lng: Double = 0.0,
        val point: Int = 1000
) {
    fun toEntity(): UserEntity {
        return UserEntity(
                id = id,
                name = name,
                lat = lat,
                lng = lng,
                point = point,
                email = email,
                password = password
        )
    }

    companion object {
        fun fromEntity(user: UserEntity): UserDto {
            return UserDto(
                    id = user.id,
                    name = user.name ?: "example",
                    point = user.point,
                    email = user.email,
                    lat = user.lat ?: 0.0,
                    lng = user.lng ?: 0.0,
                    password = user.password
            )
        }
    }
}
